### PR TITLE
Feature: Union schema compatibility

### PR DESCRIPTION
**Pull request by fivetran-catfritz**
Issue: 
- https://github.com/fivetran/dbt_app_reporting/issues/20

Confirm the following files were correctly updated automatically:
- [x] CHANGELOG
- [x] README
- [x] apple_store.yml
- [x] docs.md
- [x] joins
  - note when joining the seeded `country_codes`, `source_relation` would not be included
- [ ] ~window functions (partition by)~
- [x] 'source_relation' column added in transform models
- [x] tests

Manual updates:
- [x] Add `source_relation` to downstream models if necessary
- [x] Finish any incomplete/incorrect joins/partitions
- [x] Update tests to include `source_relation` in unique-combination-of-cols if necessary
    - May need to remove some uniqueness tests in favor of the combo test

Validation:
- Consistency: When not using unioning, tests in the folder `tests/consistency/row_counts` pass
  - It is is expected that the consistency tests will not pass when unioning
- Integrity: I was able to write 1 integrity test for `app_store_territory_source_type_report` since there wasn't much aggregation from the source, but the others were too complex to write a meaningful test.
- Spot check with seed data for unioning:
  - Screenshot 2024-07-18 at 1 06 11 PM

- Also:
  - see internal ticket

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,39 @@
...
+`usage_territory_source_type_report_identifier` | `apple_store_usage_territory_source_type_report_identifier`
+
+- If you are using the previous identifier, be sure to update to the current version!
+
+## Feature update 🎉
+- Unioning capability! This adds the ability to union source data from multiple apple_store connectors. Refer to the [README](https://github.com/fivetran/dbt_apple_store/blob/main/README.md#union-multiple-connectors) for more details.
+
+## Under the hood 🚘
+- In the source package, updated tmp models to union source data using the `fivetran_utils.union_data` macro. 
+- To distinguish which source each field comes from, added `source_relation` column in each staging and downstream model and applied the `fivetran_utils.source_relation` macro.
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:33]: This also has impacts in these end models as the `source_relation` is now a new field in all end models. We should highlight this in the feature update for the end models.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-catfritz great work on this PR! I just have a few small requests before approving this PR. Let me know if you have any questions. Thanks!

**Review by fivetran-catfritz — COMMENTED** [review]
@fivetran-joemarkiewicz I have updated, and this is ready for re-review!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM thanks for applying the changes!

**Pull request merged**